### PR TITLE
fix for run on success

### PR DESCRIPTION
**Pull request by harshil-hbo**

### CODE REVIEW

File: src/main/java/hudson/plugins/jira/JiraCreateIssueNotifier.java [hunk 1]
```diff
@@ -156,8 +156,10 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
         Result previousBuildResult = null;
         AbstractBuild<?, ?> previousBuild = build.getPreviousCompletedBuild();
 
-        if (forceNewIssue && currentBuildResult == Result.FAILURE) {
-            currentBuildResultFailure(build, listener, null, filename, vars);
+        if (forceNewIssue) {
+            if (currentBuildResult == Result.FAILURE) {
+                currentBuildResultFailure(build, listener, null, filename, vars);
+            }
         }
```
Comment (elliot-nelson) [inline src/main/java/hudson/plugins/jira/JiraCreateIssueNotifier.java]: @harshil-hbo What issue are you looking at here?  This change is actually identical to the previous code (you are already using `&&`, which is a conditional AND -- the right-hand-side of the operator is never evaluated if `forceNewIssue` is false.)  Also, even if the expression _was_ evaluated, it couldn't cause an exception -- comparing `NULL` to a value is always valid in Java.

**Comment by elliot-nelson** [comment]
I would recommend scrapping this PR, as I don't think this change is necessary.

Were you looking for a potential source for yesterday's issues in Jenkins?  If so, I think you are just looking in the wrong place -- I would check out the JiraChangeLogAnnotator.  I'll ping you with some details in Slack.

File: src/main/java/hudson/plugins/jira/JiraCreateIssueNotifier.java (hunk 1 above)
Comment (harshil-hbo) [reply src/main/java/hudson/plugins/jira/JiraCreateIssueNotifier.java]: Yes thats true. But it also means if either of them is false, it goes to else. I don't want it to go to else if `forceNewIssue` is on. So if `currentBuildResult` is Result.SUCCESS, we do nothing and return true for `perform`. Before the change, we would go to else when `currentBuildResult` == `Result.SUCCESS` not caring about `forceNewIssue`. That is a behaviour we don't want

**Comment by elliot-nelson** [comment]
Makes sense -- I think you might want to add an inline comment just to indicate why it is the way it is.  LGTM.

**Review by elliot-nelson — APPROVED** [review]

**Pull request merged**